src/data/validate.rs

```diff
@@ -0,0 +1,365 @@
+use crate::config::Mapping;
+
+/// Validate paths for entity
+pub fn validate_paths_for_entity(
+    entity: &str,
+    mappings: &Vec<Mapping>,
+    api_schema: &serde_json::Map<String, serde_json::Value>,
+) -> anyhow::Result<()> {
+    // if entity name is not set in api_schema throw an exception
+    if !api_schema.contains_key(entity) {
+        anyhow::bail!("Entity {} not found in API schema", entity);
+    }
+
+    for entry in mappings {
+        let path_mapping = match entry {
+            Mapping::ByPath(path_mapping) => path_mapping,
+            Mapping::ByScript(_) => continue,
+        };
+
+        let path = path_mapping.entity_path.split('.').collect::<Vec<_>>();
+
+        // if path starts with ? its optional
+        if path[0].ends_with('?') {
+            continue;
+        }
+
+        // check if first path element is set in the entity object in schema
+        if !api_schema[entity]["properties"][path[0]].is_object() {
```

Comment:
In general, every indexing like `api_schema[...]` can panic. The cleaner alternative is calling the `get(...)` method which returns an `Option`.

Dealing with all these untyped `serde_json::Value` values isn't super nice, as there is basically no guarantee what you will find. But on the other hand defining structure in Rust and properly parsing the entity_schema is also not an easy task.

There is one utility method `serde_json::Value::pointer`, see https://docs.rs/serde_json/latest/serde_json/enum.Value.html#method.pointer
That might be easier to use here, but unfortunatly that isn't implement on `serde_json::Map`. I implemented my own `get_by_path` (see `transform.rs`), which you should be able to use on `api_schema`. That is also used to resolve the `entity_path` in profiles.